core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -322,7 +333,12 @@ private[spark] class TaskSetManager(
       // Check for node-local tasks
       if (TaskLocality.isAllowed(locality, TaskLocality.NODE_LOCAL)) {
         for (index <- speculatableTasks if canRunOnHost(index)) {
-          val locations = tasks(index).preferredLocations.map(_.host)
+          val locations = if ( sparkLocalHostname != None ) {
```

Comment:
there are cases which we didn't cover in the scala style rules but are actually imposed "manually"....

e.g. even you remove the spaces around `=`, it still passes the scala style checks.....